main.go

```diff
@@ -179,6 +180,26 @@ func main() {
 		os.Exit(1)
 	}
 
+	var autoAnnotationConfig auto.AnnotationConfig
+	var autoAnnotation *auto.AnnotationMutators
+	if os.Getenv("DISABLE_AUTO_ANNOTATION") != "true" {
+		if err := json.Unmarshal([]byte(autoAnnotationConfigStr), &autoAnnotationConfig); err != nil {
+			setupLog.Error(err, "unable to unmarshal auto-annotation config")
+		} else {
+			autoAnnotation = auto.NewAnnotationMutators(
+				mgr.GetClient(),
+				mgr.GetAPIReader(),
+				logger,
+				autoAnnotationConfig,
+				instrumentation.NewTypeSet(
+					instrumentation.TypeJava,
+					instrumentation.TypePython,
+				),
+			)
+			go autoAnnotation.MutateAll(ctx)
```

Comment:
I think we should call this after we register the webhook etc towards the end just before actually starting the manager to minimize race conditions.

Reply:
Even if we move it after the webhook, we will still run into the race condition, now the webhook will try to annotate first before the goroutine. We could add a delay but that's going to get tricky as well

I dont have strong thoughts about either

Reply:
The webhook wont get triggered unless there was a create or update to a resource in the gap between the two. So i think thats far less likely to happen.
But agree we need something more deterministic like checking the manager is up and running via a health check or something.